sklearn/preprocessing/_data.py

```diff
@@ -1837,6 +1842,349 @@ def transform(self, X):
         return XP
 
 
+# TODO:
+# - sparse support (either scipy or own cython solution)?
+# - extrapolation (cyclic)
+class SplineTransformer(TransformerMixin, BaseEstimator):
+    """Generate univariate B-spline bases for features.
+
+    Generate a new feature matrix consisting of `n_spline` spline basis
```

Comment:
```suggestion
    Generate a new feature matrix consisting of spline basis
```

you speak about `n_splines` below after introducing n_knots and degree

Reply:
I would prefer to keep this information here. So instead of removing `n_splines`, I'll add the formula.